dotnet/src/InternalUtilities/process/ProcessExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Linq;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.SemanticKernel.Process.Runtime;
+
+internal static class ProcessExtensions
+{
+    public static KernelProcess CloneProcess(this KernelProcess process, ILogger logger)
```

Comment:
should the cloning have optional keepExistingIds bool? wondering if there may be a problem with having the same ids in multiple steps.

same with the ids of the steps

Reply:
That's a good point.  For map step, the cloned versions are ran as isolated processes / steps.  I do wonder about how changing the id might affect the edge definitions.  (I think that's less of an issue for Processes and more for Steps.)  MapStep needs to use its flavor of Clone for both (every) runtime...unclear if Clone is useful outside of that.  

Thoughts?

Reply:
yeah I'm mostly thinking about subprocess steps, I've been working with those and it gets messy since then potentially the subprocess steps would also have the same ids as the other subprocesses used as steps

Reply:
Updated to assign subprocess new id, always.

Reply:
Actually, a bit more testing showed that the id switch is invalidating edges.  Re-writing all edges to be consistent with the new id is entirely non-trivial.  I'll be happy to revisit this with you should it become problematic in the future.

Reply:
Changing Ids will be quite problematic for steps or sub processes that are defined in the "blueprint" of the process. I think we should aim to limit the cloning to the MapStep of similar steps that need to create new instances that are not defined in the blueprint.